Add debugging instructions (#1451)
jenkinsci/gitlab-plugin: CONTRIBUTING.md

See https://github.com/jenkinsci/gitlab-plugin/tree/master/src/docker/README.md

> ## Using INTELLiJ's Debugger

Reviewer: ```suggestion
## Using IntelliJ's Debugger
```

> When testing with Docker Jenkins instance, the debugger can be setup in the following way :

Reviewer: ```suggestion
When testing with a Docker Jenkins instance, the debugger can be setup in the following way:
```

* From the main menu, select Run -> Edit Configurations.
* In the Run/Debug Configurations dialog, click the Add New Configuration button `+` and select Remote JVM Debug.

> * Enter any relevant name, Host(the address of the machine where the host app will run. If running it on the same machine, it needs to be localhost. If the program is running on another machine, specify its address here) and Port(by default use 50000). The Command Line argument will automatically be setup.

Reviewer: ```suggestion
* Enter any relevant name, the Host (the address of the machine where the host app will run. If running it on the same machine, it needs to be localhost. If the program is running on another machine, specify its address here) and the Port (by default use 50000). The Command Line argument will be automatically setup.
```

* Enter Apply.

Now start your Jenkins instance and debugger and you should get something like this - `Connected to the target VM, address: 'localhost:50000', transport: 'socket'`.

> Breakpoints can now be setup to halt the debugger at the required point and understand the flow of the program.  

Reviewer: ```suggestion
Breakpoints can now be set up to halt the debugger at the required break point to understand the flow of the program.  
```
